<!--
   Copyright 2022 eve autonomy inc. All Rights Reserved.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
-->

<launch>
    <arg name="time_required_to_release_button" default="5.0" />
    <arg name="timeout_period_before_shutdown_aborts" default="10.0" />

    <node pkg="shutdown_manager" exec="shutdown_manager_node" name="shutdown_manager" output="screen">
      <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)" />
      <param name="time_required_to_release_button" value="$(var time_required_to_release_button)" />
      <param name="timeout_period_before_shutdown_aborts" value="$(var timeout_period_before_shutdown_aborts)" />
      <remap from="input/shutdown_button" to="/shutdown_button" />
      <remap from="output/state" to="/shutdown_manager/state" />
    </node>
</launch>
